src/main/java/states/ReadState.java

```diff
@@ -0,0 +1,99 @@
+package states;
+
+import static resources.Configuration.GLOBAL_CONFIG;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.SocketAddress;
+import java.net.UnknownHostException;
+
+import exceptions.InvalidPacketException;
+import formats.AckMessage;
+import formats.DataMessage;
+import formats.RequestMessage;
+import logging.Logger;
+import formats.Message.MessageType;
+import resources.ResourceManager;
+import socket.TFTPDatagramSocket;
+
+public class ReadState extends State {
+	private static final int SOCKET_TIMEOUT = 5000;
+	
+	private TFTPDatagramSocket socket;
+	private SocketAddress serverAddress;
+	private String filename;
+	
+	public ReadState(SocketAddress serverAddress, String filename, boolean isVerbose) {
+		this.serverAddress = serverAddress;
+		this.filename = filename;
+		if (isVerbose)
+			Logger.setLogLevel(Logger.LogLevel.VERBOSE);
+		else
+			Logger.setLogLevel(Logger.LogLevel.QUIET);
+	}
+
+	@Override
+	public State execute() {
+		int expBlockNum = -1;
+		try {
+			// Create the resource manager, handle IOException if failed to get resource directory
+			ResourceManager resourceManager = new ResourceManager(GLOBAL_CONFIG.CLIENT_RESOURCE_DIR);
```

Comment:
To share this with the server we'll probably need to inject this config as well as the Logger.